src/biogeophys/LunaMod.F90

```diff
@@ -472,8 +478,8 @@ subroutine Update_Photosynthesis_Capacity(bounds, fn, filterp, &
                 endif !if not C3 plants                   
          else
             do z = 1 , nrad(p)
-               jmx25_z(p, z) = 85._r8
-               vcmx25_z(p, z) = 50._r8
+               jmx25_z(p, z) = jmx_prevyr(p,z)
+               vcmx25_z(p, z) = vcmx_prevyr(p,z)
```

Comment:
Is this really a bug fix?  More broadly, why is the evolution of annual GPP so sensitive to these initial conditions? @lmbirch89 can you provide more background on this?

Reply:
@wwieder It's sensitive to these initial conditions because the evolution of Jmax and Vcmax throughout the growing season is dependent on the initial value in spring and a constraint on the amount of change these values can have in each time step. The choice here was to remove the constraint completely or initialize the values more appropriately in Spring. Basically, if the initial spring value is biased then the maximum constraint variable doesn't allow for a rapid change to more accurate values. If the maximum constraint isn't there, then the next value of Jmax and Vcmax doesn't depend on the previous value, which may in fact bee a valid assumption to make. I chose to change the spring initialization, but another choice could be to evaluate if the maximum constraint term is valid. 

If that doesn't quite make sense, we can chat again about it.

Reply:
@lmbirch89 - thank you for this explanation. This explanation makes sense to me, though I'll let others reply about any scientific considerations.

There are some technical considerations that will need to be kept in mind if we stick with this dependence on the last year's value:

- Care is needed to make sure this works reasonably for (1) cases using an older restart file where the new prevyr variables aren't present, and (2) transient cases: need to ensure that the initialization is reasonable for a crop PFT that is newly coming into existence.

- Looking quickly through accumulMod, I'm not sure if we'll be able to use any existing accumulation routines, since the nearly-relevant routines seem to assume that values are valid for all points (as long as they are "active" – i.e., not 0-area – points). Here it seems that the logic involves: for a given time, some points will be accumulating this field while others will not. It should be doable to add the right logic in accumulMod, but I just wanted to point out that it looks like we can't use this exactly as is (unless I'm missing something, which is quite possible).

I also wonder about a couple of other options that would be simpler to implement. I want to suggest these for consideration, but the final decision should be based more on what makes sense scientifically, since I don't think any of these will be *too* hard to implement:

- If I understand your current implementation, it seems to set the prevyr variables equal to something like the value from the last active day in the previous year. If that is scientifically reasonable (as opposed to actually taking the average over the previous year), then keeping this logic would be simplest. I may be misunderstanding your current implementation, though, so please feel free to correct me on this point.

- Your comment also made me wonder about an alternative approach: Would it make sense to have the change constraint depend on the number of days the given PFT has been active? So if this is the first day that it's active, then the values can change without constraint, and this constraint gradually tightens over X days. (This is similar to approaches sometimes used in parameter optimization.) This would require keeping track of the number of days this PFT has been active, but that is probably simpler than putting in place the necessary accumulation code. Again, though, this is probably only worth doing if this is equally or more justifiable scientifically than the prevyr approach.

Reply:
> Scientifically I wonder what mechanisms plants have to "remember" the previous years value for this value? It would seem unlikely that they do have a memory of the previous year -- but I don't know.

On the other hand, I could see the argument that dependence on the previous year is less about a memory in individual plants and more about capturing spatial variability in this parameter due to local adaptations. Although, with that argument, something like a 20-year average might be better so that one bad year doesn't influence the next year....

The biggest problem I see with my last suggestion is that it could lead to high sensitivity to the conditions at the start of the growing season in the given year, which also may be undesirable.

I should be quiet and let the scientists comment :-)

Reply:
I should start by saying that I also don't know anything plants, but that I'd imagine plants are just as likely to start where the'd left off from the year before, based on historical environmental conditions, as they would be to reset photosynthetic capacity to a global value?

Reply:
Plants might not have "memory" per se, but will have some starting point based on genetics and acclimation to local environments. I agree with @billsacks that a multi-year average might help to smooth out interannual variability and perhaps reduce the sensitivity that @billsacks mentioned above.

Reply:
I think I like @lmbirch89's change here. It does seems it is a bug and I think In my original version, we did use the previous year values.  Scientifically, I think it make sense as well as the plant will get to a condition specific to the local climate and not starting from a new value each year.

Reply:
@billsacks and  @lmbirch89, I agree with @billsacks 's point that the values should be from end of last year instead of the mean of last year. "If I understand your current implementation, it seems to set the prevyr variables equal to something like the value from the last active day in the previous year.".

Reply:
@xuchonggang , can you explain further why you think it should use values from the last active day of the previous year rather than the average?  If the 'goal' is to have something that tracks the local climate better than some specified value for all climates, it's not clear to me why you would use the last active value, which could be somewhat anomalous, depending on the local environmental conditions were on that particular day.  One possibility is that the answer is that it really doesn't matter too much what initial value is used, it is just better to use a value that is at least reasonable for the local climate.

Reply:
I actually have the same question. I thought the last value would be too anomalous, which is why I did a running average. However, @dlawrenncar  I think there is a good chance it doesn't matter too much as long as the values are able to adjust to the local climate and vegetation type on a grid cell.

Reply:
@lmbirch89 - I'm confused by your last comment, because my read of your current implementation is that you are using the value from the end of the last year, *not* the annual average. I'm sorry to belabor this point, but I think it's important that we're all in agreement about what your current implementation does – and so what you have actually tested – to set the stage for a discussion of how we *want* this to work.

Here is an excerpt from the code on your branch, where I have removed a bunch of code in order to focus in on the logic around `vcmx25_z` and `vcmx_prevyr`... I think I got this right, but it's possible I made a mistake somewhere:

```Fortran
  do f  =  1,fn
     p  =  filterp(f)
     ft =  patch%itype(p)     
     !----------------------------------------------------
     !store the daily mean climate conditions
     if(t_veg_day(p).ne.spval) then    !check whether it is the first day 
        if(tlai(p) > 0.0_r8 .and. lnc(p) > 0._r8)then
           if (nint(c3psn(ft)) == 1)then
              if(gpp_day(p)>0.0 )then   !only optimize if there is growth and it is C3 plants
                 do z = 1 , nrad(p)
                    vcmx_prevyr(p,z) = vcmx25_z(p,z)
                    vcmx25_z(p, z)  = vcmx25_z(p, z)+sign(1.0_r8,chg)*chg_constrn
                    vcmx_prevyr(p,z) = (vcmx_prevyr(p,z)+vcmx25_z(p,z))/2.0_r8
                 end do
              else
                 do z = 1 , nrad(p)
                    if(enzs_z(p,z)>0.5_r8) then
                       vcmx25_z(p, z) = vcmx25_z(p, z)* (1.0_r8 - max_daily_decay) 
                    end if
                 end do
              end if
              ! Note from Bill: there is no 'else' clause here, so vcmx_prevyr appears to
              ! never be set for c4 plants. Is that a problem?
           end if
        else
           do z = 1 , nrad(p)
              vcmx25_z(p, z) = vcmx_prevyr(p,z)
           end do
        end if
     end if
  end do
```

My read of this code is that, under the right conditions (tlai > 0, lnc > 0, c3psn == 1 and gpp_day > 0), vcmx_prevyr is set to the average of the previous (yesterday's?) value of vcmx25_z and the new value of vcmx25_z. So then, when vcmx_prevyr is used in the `else` at the bottom, it seems like it will be the average over the last two values of `vcmx25_z`, not the annual average. Am I missing something? (It's quite possible that I'm missing something since I don't understand the whole Luna structure.)

By the way, I also noticed from this that vcmx_prevyr never seems to be set for c4 plants, although it is used on the right hand side for c4 plants. Does that matter? There also seem to be a lot of ways that vcmx_prevyr could be used before it's set, so if we stick with this implementation, we'll need to be sure that it starts out at a reasonable value.

Reply:
The reason to use the last day's value is the continuity of vcmax, namely the amount of enzyme in the leave. During the winter or drought, enzyme might deteriorate and use last year's end value instead of the mean will show the continuity of enzyme activities.

Reply:
@billsacks you are absolutely right, and apologies again for my poor translation into the updated code version. This one is only an average of the last 2 time steps, which was commented out in other simulations that I ran, so it was a running average. I had tested a few different averages of jmax and vcmax, trying to decide if end of season or running average was best. The point about keeping the enzymes was not something I had considered about why use the final time step.

Reply:
Thanks for the clarification. I'll let others make the call about what is best scientifically here - and whether the benefits of something more complex than the current implementation justify the extra complexity.

Reply:
OK, it sounds like using the value from the end of the previous growing season will be an appropriate starting value and that this is:

- Scientifically valid
- Already tested in code @lmbirch89 used
- Easier to implement technically

IF this is accurate reading of the discussion, above, @billsacks and @ekluzek, what can be done to clarify the code (and write out appropriate information to restart files)?

Reply:
> IF this is accurate reading of the discussion above, @billsacks and @ekluzek, what can be done to clarify the code (and write out appropriate information to restart files)?

I believe @lmbirch89 already added these to restart files. In terms of code clarity, I would mainly suggest working on the variable name. Perhaps something like `vcmx_last_active_day`? Probably somebody who understands the code better can come up with a better name, but I want the name to accurately suggest that this is just a single day's value, not an annual average.

I do wonder why the current code takes the average of the last two values, rather than just taking the last value. Is there a significant benefit to taking this average? This average doesn't add much code complexity, but it does make it hard to name the variable accurately. And, since an inaccurate name can lead to confusion, I wonder if it's worth simplifying this to just be set to the most recent vcmax value rather than the average of the last two?

One other question is whether this should be set for c4 plants - or is this irrelevant for c4 plants?

I see that @lmbirch89 has initialized these to reasonable values already, in case they are used before being set, which is good. However, someone should confirm that the current initialization `vcmx_prevyr = 30`, `jmx_prevyr = 60` is what is desired, rather than using the previous hard-coded values of 50 and 85, or something else. Note that these cold start values will be used: In the first year of a cold start simulation; in the first year after restart for cases using old restart files; in the first year of existence of crop that newly appears in a grid cell in a transient case; and possibly other cases where this variable ends up getting used before being set.

Reply:
@billsacks In the simulations where I did a running average:
vcmx_prevyr(p,z) = vcmx25_z(p,z)
was commented out. I added it in to save values as I was working through what was going on and if different values should/could be used. As it is written now, it does just save the last two values. 

I am also fairly certain that LUNA was just for c3.

Reply:
@lmbirch89 - My question on this point is: For the final code, is there any value in taking the average of the last two values, or should we simply set `vcmx_last_active_day = vcmx25_z` after the new day's value of `vcmx25_z` is calculated?

Reply:
addressed in #961, but renaming these variables may be helpful